dolphin/api/v1/storages.py

```diff
@@ -180,22 +182,23 @@ def create_resource():
 
 
 @coordination.synchronized('{storage_id}')
-def _set_synced_if_ok(context, storage_id):
+def _set_synced_if_ok(context, storage_id, resource_count):
     try:
         storage = db.storage_get(context, storage_id)
     except exception.StorageNotFound:
         msg = 'Storage %s not found when try to set sync_status' \
               % storage_id
         raise exception.InvalidInput(message=msg)
     else:
-        if storage[constants.DB.DEVICE_SYNC_STATUS] != \
+        last_update = storage['updated_at']
+        current_time = datetime.now()
+        interval = (current_time - last_update).seconds / 60
+        # If last synchronization was within 30 minutes,
+        # and the sync status is not SYNCED, it means some sync task
+        # is still running, the new sync task should not launch
+        if interval < 30 and storage[constants.DB.DEVICE_SYNC_STATUS] != \
```

Comment:
plz add expiration in confiuration file, set second as unit.

Reply:
Done